tests/sympc/protocol/aby3/aby3_test.py

```diff
@@ -187,3 +188,31 @@ def test_local_decomposition_exception() -> None:
     x = ReplicatedSharedTensor()
     with pytest.raises(ValueError):
         ABY3.local_decomposition(x, "2")
+
+
+@pytest.mark.parametrize("security_type", ["semi-honest", "malicious"])
+def test_bit_decomposition_ttp(get_clients, security_type) -> None:
+    parties = get_clients(3)
+    falcon = Falcon(security_type=security_type)
+    session = Session(parties=parties, protocol=falcon)
+    SessionManager.setup_mpc(session)
+    secret = torch.tensor([[-1, 12], [-32, 45], [98, -5624]])
+    x = MPCTensor(secret=secret, session=session)
+    b_sh = ABY3.bit_decomposition_ttp(x, session)
+    ring_size = x.session.ring_size
+    tensor_type = x.session.tensor_type
+    ring_bits = get_nr_bits(ring_size)
+
+    val = 1
+    result = 0
+    for i in range(ring_bits):
+        if i != ring_bits - 1:
+            result += b_sh[i].reconstruct(decode=False).type(tensor_type) * val
+        else:
+            result += b_sh[i].reconstruct(decode=False).type(tensor_type) * (-val)
+        val *= 2
+
+    exp_res = torch.tensor(
+        [[-65536, 786432], [-2097152, 2949120], [6422528, -368574464]]
```

Comment:
why not compare decoded values?

Reply:
George asked me in the previous PR's to harcode ,decoding would make us assume the correctness of reconstruction.

Reply:
Makes sense.